packages/inferno-router/src/BrowserRouter.ts

```diff
@@ -25,11 +25,13 @@ export default class BrowserRouter extends Component<IBrowserRouterProps, any> {
   }
 
   public componentWillMount() {
-    warning(
-      !this.props.history,
-      "<BrowserRouter> ignores the history prop. To use a custom history, " +
-        "use `import { Router }` instead of `import { BrowserRouter as Router }`."
-    );
+    if (process.env.NODE_ENV !== "production") {
```

Comment:
This is not exactly what I meant, because now it will have empty CWM function defined. Better to assign it into prototype only when needed: 
See => https://github.com/infernojs/inferno/blob/dev/packages/inferno-mobx/src/Provider.ts#L43-L64